backend/tarsy/controllers/chat_controller.py

```
@@ -0,0 +1,486 @@
+            created_at_us=chat.created_at_us,
+            created_by=chat.created_by or "unknown",
+            message_count=message_count,
+        )
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Failed to get chat {chat_id}: {str(e)}", exc_info=True)
+        raise HTTPException(
+            status_code=500, detail=f"Failed to retrieve chat: {str(e)}"
+        )
```

Ruff (0.14.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)